pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,408 @@
+        system = intg.system
+        self.dtype = intg.system.backend.fpdtype
+        self.ndims = intg.system.ndims
+
+        # Underlying system elements class
+        self.elementscls = system.elementscls
+
+        # Load conduit library
+        self.conduit = ConduitWrappers()
+
+        # Expressions to plot and configs
+        self._exprs = defaultdict(dict)
+        self._fields_write = set()
+        self._fields_read = set()
+        self._config_field(cfgsect)
+        self._config_scene(cfgsect)
+        self._config_pipeline(cfgsect)
+
+        if not self._fields_read.issubset(self._fields_write):
+            raise AscentError('Not all fields used are defined')
+        
+        # Gradient pre-processing
+        self._init_gradients()
+
+        # Base directory for output and counter reset
+        basedir = self.cfg.getpath(cfgsect, 'basedir', '.', abs=True)
+
+        for scfg in self._scene_cfg.values():
+            scfg['path_gen'] = file_path_gen(basedir, scfg['image_name'],
+                                             intg.isrestart)
+            next(scfg['path_gen'])
+
+        # Set subdivision
+        self.divisor = self.cfg.getint(cfgsect, 'division', 
+                                       self.cfg.getint('solver', 'order'))
+        self.etypes_div = defaultdict(lambda: self.divisor)
+
+        # Share number of ele types per part
+        ndoms = len(system.ele_map)
+        self.dom_0 = dom_0 = comm.scan(ndoms) - ndoms
+
+        # Generate conduit blueprint of mesh
+        self.mesh_n = mesh_n = ConduitNode(self.conduit)
+
+        self._ele_regions_lin = []
+        for idx, ename, rgn in self._ele_regions:
+            if ename in ['pri', 'pyr']:
+                raise AscentError(f'{ename} elements not supported')
+            eles = system.ele_map[ename]
+
+            dpf = f'domain_{dom_0 + idx}'
+
+            shape = self._get_shape(ename, eles.nspts)
+
+            svpts = shape.std_ele(self.etypes_div[ename])
+            nsvpts = len(svpts)
+
+            mesh_op = shape.sbasis.nodal_basis_at(svpts).astype(self.dtype)
+            soln_op = shape.ubasis.nodal_basis_at(svpts).astype(self.dtype)
+            self._ele_regions_lin.append((idx, ename, rgn, mesh_op, soln_op))
+
+            mesh_n[f'{dpf}/state/domain_id'] = rank
+            mesh_n[f'{dpf}/coordsets/coords/type'] = 'explicit'
+            mesh_n[f'{dpf}/topologies/mesh/coordset'] = 'coords'
+            mesh_n[f'{dpf}/topologies/mesh/type'] = 'unstructured'
+            mesh_n[f'{dpf}/topologies/mesh/elements/shape'] = ename
+
+            xd = np.einsum('ik,k...', mesh_op, 
+                           eles.ploc_at_np('linspts')[..., rgn])
+            for l , x in zip('xyz', xd.reshape(eles.ndims, -1)):
+                mesh_n[f'{dpf}/coordsets/coords/values/{l}'] = x
+
+            subdvcls = subclass_where(BaseShapeSubDiv, name=ename)
+            sconn = subdvcls.subnodes(self.divisor)
+            neles_rgn = xd.shape[1]
```

Comment:
Only used in one place.

Reply:
yep, I did it because of line length. I'll just add a new line